File: structs/pilhajv/Conversor.java
public class Conversor{
    public static void main(String args[]){
        int numero = Integer.parseInt(args[0]);
        Pilha p = new Pilha();

        int resto;
        // fase 1 = empilhamento
        while (numero != 0){
            resto = numero % 2; 
            p.push(resto);
            numero = numero / 2;

        }
        //fase 2 = exibir resto
        while (!p.isEmpty()){
            resto = p.pop(); // remove da pilha
            System.out.print(resto);
        }
        System.out.println("\nFim do programa");

    }
}

// 172(10) = 10101100(2)

File: structs/pilhac/conversor.c
#include "pilha.h"
#include <stdio.h>

int main (int argc, char *argv[]){
    int valor, resto;
    struct Pilha p;
    create(&p);
    printf("Digite um valor :");
    scanf("%d", &valor);

    while(valor != 0){
        resto = valor % 2;
        push(&p, resto);
        valor = valor / 2;
    }

    while (!isEmpty(p)){
        resto = pop(&p);
        printf("%d", resto);
    }

    printf("\n\nFim do programa\n\n");
    return 0;
}
